Source/monster.h

```diff
@@ -167,107 +166,115 @@ struct CMonster {
 };
 
 struct Monster { // note: missing field _mAFNum
-	const char *mName;
-	CMonster *MType;
-	const MonsterData *MData;
-	std::unique_ptr<uint8_t[]> uniqueTRN;
-	AnimationInfo AnimInfo;
-	int _mgoalvar1;
-	int _mgoalvar2;
-	int _mgoalvar3;
-	int _mVar1;
-	int _mVar2;
-	int _mVar3;
-	int _mmaxhp;
-	int _mhitpoints;
-	uint32_t _mFlags;
+	const char *name;
+	CMonster *type;
+	const MonsterData *data;
+	std::unique_ptr<uint8_t[]> uniqueMonsterTRN;
+	AnimationInfo animInfo;
+	/** Specifies current goal of the monster */
+	monster_goal goal;
+	/** Specifies monster's behaviour regarding moving and changing goals. */
+	int goalGeneral;
+	/** @brief Specifies turning direction for @p RoundWalk in most cases.
+	 *  Used in custom way by @p FallenAi, @p SnakeAi, @p M_FallenFear and @p FallenAi.
+	 */
+	int goalTurning;
+	/** @brief Controls monster's behaviour regarding special actions.
+	 *  Used only by @p ScavengerAi and @p MegaAi.
+	 */
+	int goalSpecialAction;
+	int var1;
+	int var2;
+	int var3;
+	int maxHp;
+	int hitPoints;
+	uint32_t flags;
 	/** Seed used to determine item drops on death */
-	uint32_t _mRndSeed;
+	uint32_t rndItemSeed;
 	/** Seed used to determine AI behaviour/sync sounds in multiplayer games? */
-	uint32_t _mAISeed;
-
-	uint16_t mExp;
-	uint16_t mHit;
-	uint16_t mHit2;
-	uint16_t mMagicRes;
-	_speech_id mtalkmsg;
+	uint32_t aiSeed;
+	uint16_t exp;
+	uint16_t hit;
+	uint16_t hit2;
+	uint16_t magicRes;
+	_speech_id talkMsg;
 	ActorPosition position;
 
 	/** Usually corresponds to the enemy's future position */
 	WorldTilePosition enemyPosition;
-	uint8_t _mMTidx;
-	MonsterMode _mmode;
-	monster_goal _mgoal;
-	uint8_t _pathcount;
+	uint8_t idx;
+	MonsterMode mode;
+	uint8_t pathCount;
 	/** Direction faced by monster (direction enum) */
-	Direction _mdir;
-	/** The current target of the monster. An index in to either the plr or monster array based on the _meflag value. */
-	uint8_t _menemy;
+	Direction dir;
+	/** The current target of the monster. An index in to either the player or monster array based on the _meflag value. */
+	uint8_t enemy;
 
 	/**
 	 * @brief Contains information for current animation
 	 */
-	bool _mDelFlag;
+	bool delFlag;
 
-	_mai_id _mAi;
-	uint8_t _mint;
-	uint8_t _msquelch;
-	uint8_t _uniqtype;
-	uint8_t _uniqtrans;
-	int8_t _udeadval;
-	int8_t mWhoHit;
-	int8_t mLevel;
-	uint8_t mMinDamage;
-	uint8_t mMaxDamage;
-	uint8_t mMinDamage2;
-	uint8_t mMaxDamage2;
-	uint8_t mArmorClass;
+	_mai_id ai;
+	/** Specifies monster behaviour across various actions */
+	uint8_t behaviour;

```

Comment:
Adding descriptions of how this is used like you've done for the goal vars would be great. This appears to be used as an "aggressiveness threshold", to determine how much damage the acid puddle does when left behind after a spitter attack, to determine how much health fallen gain when recovering, maybe more?

Reply:
Preferably keep it as `inteligence`, behaviour is both very broad, not more explanatory and it changes what people that are already used to the code expects.